diagnostic.md

```diff
@@ -40,31 +49,39 @@ like to show all movies favorited by a profile on
 `http://localhost:3000/profiles/1`
 
   ```md
-    # < Your Response Here >
+The purpose is to provide the data we are trying to attain from the server
   ```
 
   ```rb
   class ProfileSerializer < ActiveModel::Serializer
+    attributes :id, :movies, :title, :length, :favorites
+
+    def favorites
+      object.favorites.all
+    end
   end
   ```
 
 1.  What would the command be to _scaffold_ out a **join table** for Favorites from
 the above `Movies` and `Profiles`.
 
   ```sh
-    # < Your Response Here >
+    bin/rails generate favorites profile_id:references movie_id:references title:string length:integer
   ```
 
 1.  What is `Dependent: Destroy` and where/why would we use it?
 
   ```md
-    # < Your Response Here >
+    Dependent destroy is when you delete a child row from a parent entity. We would use this to delete a row from our join table.
```

Comment:
`dependent: :destroy` will cause the deletion of an object to also delete all associated objects.  For example, if a `User` model has `has_many :favorites, dependent: :destroy`, then deleting a particular user will also delete all rows in the `favorites` table where `user_id` refers to the `id` of the user being deleted.